repos/spack_repo/builtin/packages/py_gosam/package.py

```diff
@@ -40,11 +40,8 @@ class PyGosam(Package):
     depends_on("qgraf", type="run")
     depends_on("gosam-contrib", type="link")
     depends_on("python@3:", type=("build", "run"))
+    depends_on("py-setuptools", type="build", when="^python@3.12:")
```

Comment:
```suggestion
    depends_on("py-setuptools", type="build")
```

So this then @wdconinc?

Reply:
I think so? There might be some version limits. I think py-setuptools only started to provide distutils with something like v74 or so. Can't find it from my phone though.

Reply:
Pip has always silently replaced distutils imports with setuptools. We should always depend on setuptools.

Reply:
Merged on the expert recommendation of @adamjstewart :-)